### PR TITLE
Blokh/feat/trans valid interface

### PR DESCRIPTION
**Pull request by Blokh**
### Description
Added the functionality of transformation and validation interfaces for plugins

### CODE REVIEW

File: packages/workflow-core/src/lib/utils/context-transformers/types.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+export type TTransformationLogic = string;
+export abstract class BaseContextTransformer {
+  abstract name: string;
+  type = 'context-transformer';
+  context: TContext;
+  transformationLogic: TTransformationLogic;
+  options: {};
+
+  protected constructor(context: TContext, transformationLogic: TTransformationLogic, options: {}) {
```
Comment (yairper) [inline packages/workflow-core/src/lib/utils/context-transformers/types.ts:11]: 1. it looks like an interface with a constructor,
2. that's a bad practice to have a constructor in the base class as one of the things which separates  implementations is the constructor itself
3. also it makes the code less readable as you need to go to the base class every time you wanna understand the child implementation

File: packages/workflow-core/src/lib/utils/context-validator/json-schema-validator.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+export class JsonSchemaValidator extends BaseSchemaValidator {
+  name = "json-schema-validator";
+
+  validate(): TSchemaValidatorResponse {
+    const ajv = new Ajv(this.options);
+    const validator = ajv.compile(this.validationLogic);
+
+    if (!validator(this.data)) {
+      return Promise.resolve({ isValid: false, errorMessage: this.errorMessage || 'Bad Request' });
```
Comment (yairper) [inline packages/workflow-core/src/lib/utils/context-validator/json-schema-validator.ts:12]: I guess that should be Promise.reject, in that case `isValid: false` is pretty obvious, just return the message...

File: packages/workflow-core/src/lib/utils/context-transformers/qj-transformer.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { run } from 'node-jq';
+import { BaseContextTransformer, TContext } from './types';
+
+export class JQTransformer extends BaseContextTransformer {
+  name = 'jq-transformer';
+
+  transform(): any {
```
Comment (yairper) [inline packages/workflow-core/src/lib/utils/context-transformers/qj-transformer.ts:7]: given the interface it returns a promise not any

File: packages/workflow-core/src/lib/utils/types.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { JQTransformer } from './context-transformers/qj-transformer';
+import { JsonSchemaValidator } from './context-validator/json-schema-validator';
+import { AnyRecord } from '@ballerine/common';
+
+export type TContext = AnyRecord;
+
+export type TTransformers = JQTransformer;
```
Comment (yairper) [inline packages/workflow-core/src/lib/utils/types.ts:7]: not in use

File: packages/workflow-core/src/lib/utils/types.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+export type TValidators = JsonSchemaValidator;
```
Comment (yairper) [inline packages/workflow-core/src/lib/utils/types.ts:8]: not in use

**Review by yairper — CHANGES_REQUESTED** [review]

**Comment by yairper** [comment]
we need to talk about such PRs as this code 
1. does not implement any user scenario and we only imagine how allegedly it gonna be used, which could change
2. we are going to merge code which does not add any increment to the product we are at a risk of adding dead/wrong code

File: packages/workflow-core/src/lib/utils/context-validator/json-schema-validator.ts (hunk 2 above)
Comment (Blokh) [reply packages/workflow-core/src/lib/utils/context-validator/json-schema-validator.ts:12]: I somewhat disagree.
validator should return is valid or not. the validation should be on that field.

File: packages/workflow-core/src/lib/utils/types.ts (hunk 4 above)
Comment (Blokh) [reply packages/workflow-core/src/lib/utils/types.ts:7]: this is what you should be using for transformation and validation

File: packages/workflow-core/src/lib/utils/types.ts (hunk 5 above)
Comment (Blokh) [reply packages/workflow-core/src/lib/utils/types.ts:8]: this is what you should be using for transformation and validation

File: packages/workflow-core/src/lib/utils/context-transformers/types.ts (hunk 1 above)
Comment (Blokh) [reply packages/workflow-core/src/lib/utils/context-transformers/types.ts:11]: yeah inheritance has its issues.

but for now, it is built as intendant. maybe later we'll have to change the constructor.
later on, if the abstract constructor will require additional functionalities/requirements by any other "transformer". we'll update the logic. for now it works as intendant

File: packages/workflow-core/src/lib/utils/context-transformers/types.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    this.context = context;
+    this.transformationLogic = transformationLogic;
+    this.options = options;
+  }
+
+  abstract transform(): Promise<any>;
```
Comment (Blokh) [inline packages/workflow-core/src/lib/utils/context-transformers/types.ts:17]: pass the necessary information for transformation

File: packages/workflow-core/src/lib/utils/context-validator/json-schema-validator.ts [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import { BaseSchemaValidator, TSchemaValidatorResponse } from './types';
+import Ajv from 'ajv';
+
+export class JsonSchemaValidator extends BaseSchemaValidator {
+  name = "json-schema-validator";
+
+  validate(): TSchemaValidatorResponse {
```
Comment (Blokh) [inline packages/workflow-core/src/lib/utils/context-validator/json-schema-validator.ts:7]: async

**Pull request merged**